### ISSUE TITLE
Relative repo paths in manifest

### ISSUE THREAD

**Issue by GunnarFarneback** [JuliaLang/Pkg.jl#2247]
This is a simplified reproducer showing two issues with relative paths in `Manifest.toml`.
Start with
```
mkdir /tmp/testdir
cd /tmp/testdir
mkdir empty_depot
git clone https://github.com/JuliaLang/Example.jl.git
cd Example.jl/
mkdir test/reference_environment
julia --project=test/reference_environment
    using Pkg
    Pkg.add(PackageSpec(name="Example", path=".", rev="3c440fe40c6e37c3ad76542f2c9f010cac984dac"))
    exit()
```
The resulting `test/reference_environment/Manifest.toml` becomes
```
[[Example]]
git-tree-sha1 = "4da5b9200d71b81738241f5213c811acc074aa6c"
repo-rev = "3c440fe40c6e37c3ad76542f2c9f010cac984dac"
repo-url = "../.."
uuid = "7876af07-990d-54b4-ab0e-23690620f79a"
version = "0.5.3"
```
The `repo-url` is the relative path from the project, which is fine.

**Issue 1**: If this is done on WIndows, the `repo-url` gets a backslash instead of a forward slash, which is not fine when you try to use the Manifest on another platform, e.g. for CI.

Next emulate instantiating this Manifest in a clean CI (it's assumed `Example.jl` is cloned in the CI, so we continue where we are but with an empty depot):
```
JULIA_DEPOT_PATH=../empty_depot julia --project=test/reference_environment
    using Pkg
    Pkg.instantiate()
```
**Issue 2**: Despite recording a relative path from the project in the manifest, `instantiate` tries to use it relative to the current working directory:
```
ERROR: git repository not found at `../..`
[...]
caused by: GitError(Code:ENOTFOUND, Class:Repository, could not find repository from '/tmp')
[...]
```
Interestingly, with the incorrect manifest recorded on Windows, it is interpreted as a path relative to the project:
```
ERROR: Did not find path `/tmp/testdir/Example.jl/test/reference_environment/..\..` for `Example [7876af07]`
[...]
```

**Comment by GunnarFarneback** [comment]
Issue 2 is addressed in #2249.

Issue 1 requires switching the slash direction somewhere. Possible candidates are
1. In [`relative_project_path`](https://github.com/JuliaLang/Pkg.jl/blob/6f6d60017b389ae87a037f0eb925e3e13a2c73f1/src/Types.jl#L408).
2. After one [specific call](https://github.com/JuliaLang/Pkg.jl/blob/6f6d60017b389ae87a037f0eb925e3e13a2c73f1/src/Types.jl#L501) to `relative_project_path`.
3. When [saving the manifest](https://github.com/JuliaLang/Pkg.jl/blob/6f6d60017b389ae87a037f0eb925e3e13a2c73f1/src/manifest.jl#L202), in the same way as is done for `path`, but maybe testing `isurl` instead of `isabspath`.

I don't have enough insights to determine which approach is better or what negative side effects there may be. Please advise.

**Comment by GunnarFarneback** [comment]
Approach 3 has been implemented in #2265.
